ddtrace/span.py

```diff
@@ -185,13 +185,20 @@ def to_dict(self):
             'error': self.error,
         }
 
-        # a common mistake is to set the error field to a boolean instead of an
-        # int. let's special case that here, because it's sure to happen in
-        # customer code.
+        # a common mistake is to set some fields without the proper type
+        # and this clashes with the msgpack encoding; to handle that we
+        # convert the field to the right type before encoding
+
+        # error must be an int
         err = d.get('error')
         if err and type(err) == bool:
             d['error'] = 1
 
+        # resource must be a string
+        resource = d.get('resource')
+        if resource and not isinstance(resource, string_type):
+            d['resource'] = stringify(resource)
```

Comment:
It's already a recovery step because users have to set the `resource` as a string. It doesn't help users raising an exception because it will break their application if they don't catch that kind of issue everywhere in their code. Dropping entirely the span is another option but we're already doing that (and it still hides the issue).

Forcing the conversion to a string makes the mistake visible in the UI (if it's a mistake).